torch/nn/parallel/distributed.py

```diff
@@ -388,6 +390,9 @@ def __init__(
             "doesn't have any parameter that requires a gradient."
         )
 
+        if device_ids is not None and len(device_ids) > 1:
+            raise ValueError("device_ids can only be None or contain a single element.")
```

Comment:
Could we add some more details here? i.e. Should we mention that SPMD in PyTorch 1.9+ is retired, and link to a GH issue for users to ping if they don't have a workaround for their use case?

Reply:
I am afraid this will include too much details in the error message. Since we don't see any SPMD user at this time, we can keep the error message as simple as necessary at this time. If we see users ask questions about this in the future, then we can add more details by that time.